src/compiler/checker.ts

```diff
@@ -19287,13 +19287,21 @@ namespace ts {
             }
 
             function structuredTypeRelatedTo(source: Type, target: Type, reportErrors: boolean, intersectionState: IntersectionState): Ternary {
+                const saveErrorInfo = captureErrorCalculationState();
+                const result = structuredTypeRelatedToWorker(source, target, reportErrors, intersectionState, saveErrorInfo);
+                if (result) {
+                    resetErrorInfo(saveErrorInfo);
+                }
+                return result;
+            }
+
+            function structuredTypeRelatedToWorker(source: Type, target: Type, reportErrors: boolean, intersectionState: IntersectionState, saveErrorInfo: ReturnType<typeof captureErrorCalculationState>): Ternary {

```

Comment:
`saveErrorInfo` is kept because there are still some calls which explicitly reset it in order to hide certain errors that would be less helpful than others.